main.go

```diff
@@ -110,14 +155,153 @@ func runMain() error {
+	err = updateChangelog(changelogFilepath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while updating the changelog.")
+	}
+
+	fmt.Println("Committing changes locally...")
+	// Commit pre release changes
+	err = worktree.AddWithOptions(&git.AddOptions{All: true})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to add release changes to staging area.")
+	}
+	commitMsg := fmt.Sprintf("Finalize changes for release version '%s'", nextReleaseVersion.String())
+	_, err = worktree.Commit(commitMsg, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  name,
+			Email: email,
+			When:  time.Now(),
+		},
+	})
+
+	releaseTag := nextReleaseVersion.String()
+	vReleaseTag := fmt.Sprintf("v.%s", nextReleaseVersion.String())
+
+	undoReleaseTag := true
+	defer func() {
+		if undoReleaseTag {
+			// git tag -d
+			err = repository.DeleteTag(releaseTag)
+			if err!= nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo tag '%s'. Please run 'git tag -d %s' to delete the tag manually.", releaseTag, err)
+			}
+			err = repository.DeleteTag(vReleaseTag)
+			if err != nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo tag '%s'. Please run 'git tag -d %s' to delete the tag manually.", vReleaseTag, err)
+			}
+		}
+	}()
+
+	fmt.Println("Setting next release version tag...")
+	// Set next release version tag
+	head, err := repository.Head()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to get the ref to HEAD of the local repository.")
+	}
+	_, err = repository.CreateTag(releaseTag, head.Hash(), &git.CreateTagOptions{
+		Message: releaseTag,
+	})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to create a git tag for the next release version.")
+	}
+	_, err = repository.CreateTag(vReleaseTag, head.Hash(), &git.CreateTagOptions{
+		Message: vReleaseTag,
+	})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to create a git tag for the next release version.")
+	}
+
+	fmt.Println("Pushing release changes to master...")
+	pushCommitOpts := &git.PushOptions{Auth: gitAuth, RemoteName: originRemoteName}
+	err = repository.Push(pushCommitOpts)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while pushing release changes to origin remote.")
+	}
+
+	undoPushToMaster := true
+	defer func() {
+		if undoPushToMaster {
+			logrus.Errorf("ACTION REQUIRED: Error occurred pushing tag '%s' to master. Must undo the push to origin master manually. Follow this tutorial: 'LINK TO INSTRUCTIONS TO UNDO PUSH.'", nextReleaseVersion.String(), err)
+		}
+	}()
+
+	fmt.Println("Pushing release tag to master...")
+	pushTagOpts := &git.PushOptions{
+		Auth:       gitAuth,
+		RemoteName: originRemoteName,
+		RefSpecs:   []config.RefSpec{config.RefSpec(tagRefSpec)},
```

Comment:
Oo we can actually do better by pushing exactly the two tags that we created. This means that if the user has local-only tags, we don't force them to push them.

The way I'd do it is push the `X.Y.Z` tag, and then do a best-effort push of the `vX.Y.Z` tag (which we don't really care about if it fails), and just print a warning if the vX.Y.Z fails without failing the entire function